### PR TITLE
[Project] Change the default DotNetProject platform to anycpu

### PR DESCRIPTION
**Pull request by sgmunn**
Fixes VSTS 934343
[CSharp] Bad CPU type in executable trying to Debug/Run a default template of a CSharp project in catalina
https://vsmac.dev/934343

### CODE REVIEW

File: main/src/core/MonoDevelop.Core/MonoDevelop.Projects/ProjectService.cs [hunk 1]
```diff
@@ -53,7 +53,7 @@ public class ProjectService
 
 		TargetFramework defaultTargetFramework;
 		
-		string defaultPlatformTarget = "x86";
+		string defaultPlatformTarget = "anycpu";
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Projects/ProjectService.cs]: This isn't going to fix projects retroactively.

https://github.com/mono/monodevelop/blob/ad32b280b6e7d2fdfa1ba99920d72da11c4d890d/main/src/core/MonoDevelop.Core/MonoDevelop.Core.Assemblies/AssemblyUtilities.cs#L93-L96

We should apply a change here to return x64 on >= Catalina.

**Review by mrward — APPROVED** [review]

File: main/src/core/MonoDevelop.Core/MonoDevelop.Projects/ProjectService.cs (hunk 1 above)
Comment (sgmunn) [reply main/src/core/MonoDevelop.Core/MonoDevelop.Projects/ProjectService.cs]: this fix wasn't intending to fix things retrospectively. I'm not sure I'm following what you mean here, doesn't the code you linked do what you're suggesting, or is that an example only?

Comment (Therzok) [reply main/src/core/MonoDevelop.Core/MonoDevelop.Projects/ProjectService.cs]: The code set preferred 32 bit to run as 64bit. We should make it return 64bit only on mac+.

Maybe if it requests specific 32bit, warn once per project?

**Comment by sgmunn** [comment]
Pretty sure the Microsoft.VSForMac.Tests.AzureFunctionsTemplateTests.TestCreateBuildHttpTrigger test is unrelated, it has been timing out on many different PRs

**Comment by Therzok** [comment]
Old projects will still fail to run on Catalina. We need to either force 64bit ness or provide guidance to migrate.

File: main/src/core/MonoDevelop.Core/MonoDevelop.Projects/ProjectService.cs (hunk 1 above)
Comment (Therzok) [reply main/src/core/MonoDevelop.Core/MonoDevelop.Projects/ProjectService.cs]: Filed https://devdiv.visualstudio.com/DevDiv/_workitems/edit/960622

**Comment by mrward** [comment]
@monojenkins backport release-8.2

**Pull request merged**